test/battle/move_effect/protect.c

```diff
@@ -563,3 +563,57 @@ DOUBLE_BATTLE_TEST("Crafty Shield does not protect against moves that target all
         MESSAGE("The opposing Sunflora's Defense rose!");
     }
 }
+
+SINGLE_BATTLE_TEST("Protect: Quick Guard, Wide Guard and Crafty Shield don't max move demage", s16 damage)
+{
+    s16 dmg[2];
+    u32 move;
+
+    PARAMETRIZE { move = MOVE_WIDE_GUARD; }
+    PARAMETRIZE { move = MOVE_QUICK_GUARD; }
+    PARAMETRIZE { move = MOVE_CRAFTY_SHIELD; }
+
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE, gimmick: GIMMICK_DYNAMAX); }
+        TURN { MOVE(player, MOVE_TACKLE); MOVE(opponent, move); }
+    } SCENE {
+        HP_BAR(opponent, captureDamage: &dmg[0]);
+        HP_BAR(opponent, captureDamage: &dmg[1]);
+    } FINALLY {
+        EXPECT_EQ(dmg[0], dmg[1]);
+    }
+}
+
+SINGLE_BATTLE_TEST("Protect: Quick Guard, Wide Guard and Crafty Shield don't reduce z move demage", s16 damage)
+{
+    bool32 protected;
+    u32 move;
+
+    PARAMETRIZE { protected = TRUE; move = MOVE_WIDE_GUARD; }
+    PARAMETRIZE { protected = FALSE; move = MOVE_WIDE_GUARD; }
+
+    PARAMETRIZE { protected = TRUE; move = MOVE_QUICK_GUARD; }
+    PARAMETRIZE { protected = FALSE; move = MOVE_QUICK_GUARD; }
+
+    PARAMETRIZE { protected = TRUE; move = MOVE_CRAFTY_SHIELD; }
+    PARAMETRIZE { protected = FALSE; move = MOVE_CRAFTY_SHIELD; }
+
+    GIVEN {
+        ASSUME(GetMoveType(MOVE_TACKLE) == TYPE_NORMAL);
+        ASSUME(GetMoveEffect(MOVE_PROTECT) == EFFECT_PROTECT);
```

Comment:
```suggestion
        ASSUME(GetMoveEffect(move) == EFFECT_PROTECT);
```
I'm guessing that it's supposed to check for the current move being used.

Reply:
the protect assume should just be removed. all of the protect moves have an assumption at the top of the file